NEWS.md

```diff
@@ -93,6 +93,7 @@ function calls. (#850, #851, @renkun-ken)
   * Each linter can have multiple tags
   * New function `available_linters()` to list available linters and their tags 
     This feature is extensible by package authors providing add-on linters for {lintr}.
+  * New function `with_tags()` to help build a list of linters using tags.
```

Comment:
I think the name here only makes sense if you're very used to `with_defaults()`. but since we basically want to supersede that usage, we should start from scratch thinking about the name here.

```
linters_with_tags()
select_linters_with_tags()
get_linters_by_tag()
```

something along those lines?

Reply:
Bear in mind the most common place for this is in a `.lintr` file:

```
linters: with_tags(tags = c("default", "package_development"))
```

Maybe `linters_with_tags()` would be fine, but neither get nor select correctly describe the function, since it actively constructs linters from a factory.

Reply:
I am liking `linters_with_tags()` the most